File: Ems Backend/src/main/java/com/cms/employeemanagement/service/OffboardingService.java
package com.cms.employeemanagement.service;

import java.util.List;

import com.cms.employeemanagement.dto.OffboardingDto;

public interface OffboardingService {

    OffboardingDto createOffboarding(OffboardingDto offboardingDto);

    OffboardingDto updateOffboarding(Long id, OffboardingDto offboardingDto);

    void deleteOffboarding(Long id);

    List<OffboardingDto> getAllOffboarding();

    OffboardingDto getOffboardingById(Long id);

    List<OffboardingDto> searchByEmployeeName(String employeeName);

    List<OffboardingDto> getByStatus(String status);

    OffboardingDto completeOffboarding(Long id);

}

File: Frontend - Ems/src/services/offboardingService.js
import api from "../api/axios";

// Get All Offboarding Records
export const getAllOffboarding = () =>
    api.get("/offboarding");

// Get Offboarding By Id
export const getOffboardingById = (id) =>
    api.get(`/offboarding/${id}`);

// Create Offboarding
export const createOffboarding = (offboardingData) =>
    api.post("/offboarding", offboardingData);

// Update Offboarding
export const updateOffboarding = (id, offboardingData) =>
    api.put(`/offboarding/${id}`, offboardingData);

// Delete Offboarding
export const deleteOffboarding = (id) =>
    api.delete(`/offboarding/${id}`);

// Search Employee
export const searchEmployee = (employeeName) =>
    api.get(`/offboarding/search/${employeeName}`);

// Filter By Status
export const getByStatus = (status) =>
    api.get(`/offboarding/status/${status}`);

// Complete Offboarding
export const completeOffboarding = (id) =>
    api.put(`/offboarding/${id}/complete`);